server/src/internalClusterTest/java/org/opensearch/search/sort/SimpleSortIT.java

```diff
@@ -198,45 +198,49 @@ public void testSimpleSorts() throws Exception {
         }
         refresh();
 
-        // STRING script
         int size = 1 + random.nextInt(10);
+        final SearchResponse[] searchResponse = new SearchResponse[1];
+        // STRING script
 
         Script script = new Script(ScriptType.INLINE, CustomScriptPlugin.NAME, "doc['str_value'].value", Collections.emptyMap());
 
-        SearchResponse searchResponse = client().prepareSearch()
-            .setQuery(matchAllQuery())
-            .setSize(size)
-            .addSort(new ScriptSortBuilder(script, ScriptSortType.STRING))
-            .get();
+        int finalSize = size;
+        assertBusy(() -> {
+            searchResponse[0] = client().prepareSearch()
+                .setQuery(matchAllQuery())
+                .setSize(finalSize)
+                .addSort(new ScriptSortBuilder(script, ScriptSortType.STRING))
+                .get();
+            assertHitCount(searchResponse[0], 10);
+        });
 
-        assertHitCount(searchResponse, 10);
-        assertThat(searchResponse.getHits().getHits().length, equalTo(size));
+        assertThat(searchResponse[0].getHits().getHits().length, equalTo(size));
         for (int i = 0; i < size; i++) {
-            SearchHit searchHit = searchResponse.getHits().getAt(i);
+            SearchHit searchHit = searchResponse[0].getHits().getAt(i);
```

Comment:
Why this change is needed ?

Reply:
To allow for searchResponse to be effectively final within the assertBusy scope. 

An alternative would be to move this loop into the assertBusy scope as well and redefine the next searchResponse outside the scope.

Reply:
Yeah, we can move entire fetch response and verification under single assertBusy.